src/uct/ib/ud/accel/ud_mlx5.c

```diff
@@ -771,23 +771,23 @@ int uct_ud_mlx5_ep_is_connected(const uct_ep_h tl_ep,
     uint32_t qp_num;
     union ibv_gid *rgid;
 
-    if (!uct_ud_ep_is_same_dest_ep(&ep->super, params)) {
+    iface_addr = (uct_ud_iface_addr_t*)params->iface_addr;
```

Comment:
where do we check that iface_addr is valid? 
can't uct_ud_ep_is_connected_to_addr read it from params->iface_addr after checking flags?